tests/Tests/Api/UserApiTest.php

```diff
@@ -0,0 +1,331 @@
+{
+    private readonly UserFixture $fixture;
+
+    private ApiTestClient $testClient;
+
+    protected function setUp(): void
+    {
+        $baseUrl = getenv('OPENEMR_BASE_URL_API', true) ?: 'https://localhost';
+
+        $this->testClient = new ApiTestClient($baseUrl, false);
+        $this->testClient->setAuthToken(ApiTestClient::OPENEMR_AUTH_ENDPOINT);
+
+        $this->fixture = new UserFixture();
+        $this->fixture->load();
+    }
+
+    protected function tearDown(): void
+    {
+        $this->fixture->removeFixtureRecords();
+    }
+
+    /**
+     * @see UserRestController::post()
+     * @see UserService::insert()
+     */
+    #[Test]
+    public function postEmptyTest(): void
+    {
+        $response = $this->testClient->request('POST', '/apis/default/api/user');
+        $this->assertEquals(400, $response->getStatusCode());
+
+        $json = json_decode($response->getBody()->getContents(), true, 512, \JSON_THROW_ON_ERROR);
+        $this->assertCount(2, $json['validationErrors']);
+        $this->assertCount(0, $json['internalErrors']);
+        $this->assertArrayNotHasKey('id', $json['data']);
+        $this->assertArrayNotHasKey('uuid', $json['data']);
+    }
+
+    /**
+     * @see UserRestController::post()
+     * @see UserService::insert()
+     */
+    #[Test]
+    public function postNotCompleteTest(): void
+    {
+        $response = $this->testClient->request('POST', '/apis/default/api/user', [], [
+            'fname' => 'Igor',
+        ]);
+        $this->assertEquals(400, $response->getStatusCode());
+
+        $json = json_decode($response->getBody()->getContents(), true, 512, \JSON_THROW_ON_ERROR);
+        $this->assertCount(1, $json['validationErrors']);
+        $this->assertCount(0, $json['internalErrors']);
+        $this->assertArrayNotHasKey('id', $json['data']);
+        $this->assertArrayNotHasKey('uuid', $json['data']);
+    }
+
+    /**
+     * @see UserRestController::post()
+     * @see UserService::insert()
+     */
+    #[Test]
+    public function postSuccessMinTest(): void
+    {
+        $response = $this->testClient->request('POST', '/apis/default/api/user', [], [
+            'fname' => 'Igor',
+            'lname' => 'Mukhin',
+        ]);
+        $this->assertEquals(201, $response->getStatusCode());
+
+        $json = json_decode($response->getBody()->getContents(), true, 512, \JSON_THROW_ON_ERROR);
+        $this->assertCount(0, $json['validationErrors']);
+        $this->assertCount(0, $json['internalErrors']);
+        $this->assertNotEmpty($json['data']['id']);
+        $this->assertNotEmpty($json['data']['uuid']);
+    }
+
+    /**
+     * @see UserRestController::post()
+     * @see UserService::insert()
+     */
+    #[Test]
+    public function postSuccessMaxTest(): void
+    {
+        $data = [
+            'title' => 'Dr.',
+            'fname' => 'Aisha',
+            'mname' => 'M.',
+            'lname' => 'Patel',
+            'federaltaxid' => '45-334-1122',
```

Comment:
To avoid tripping sensitive data filters, use a known invalid tax id. The the IRS test SSN 999-99-9999 is a good options here. 

Similarly, for NPI, we have an example from a SAS test case.
Valid NPI: 1003802901 Invalid NPI: 1234567890 (fails the check digit validation) 

For testing an Employer Identification Number (EIN), use a nine-digit number that starts with a known invalid two-digit prefix. These include 00, 07, 08, 09, 17, 18, 19, 28.
You can also use a string of identical digits, as these are invalid (i.e. 88-8888888), however do not use 77-7777777.